src/lib/SurveyQuestion.svelte

```diff
@@ -163,26 +196,70 @@
   </style>
   
   {#if node.type === 'list'}
-    <ul
-      id={qname}
-      class:multi={multi && !className}
-      class:single={!multi && !className}
-      class={className}>
-      {#each node.children as child, i}
-        <!-- svelte-ignore a11y-click-events-have-key-events -->
-        <!-- svelte-ignore a11y-no-noninteractive-element-to-interactive-role -->
-        <li
-          on:click={() => check(i)}
-          class:checked={checked[i]}
-          role="checkbox"
-          aria-checked={checked[i]}>
-          <SurveyNode
-            node={{ ...child.children[0], type: 'text' }}
-            {context}
-            {next} />
-        </li>
-      {/each}
-    </ul>
+    {#if matrix !== undefined && matrix.length > 0}
+      <ul
+        id={qname}
+        class:matrix={matrix && !className}
+        style="--size:{matrix ? matrix.length + 1 : 0}"
+        class={className}>
+        <div class="matrixli">
+          <div class="matrixTxt">
+            <p></p>
+          </div>
+          {#each matrix as matrixLeg, i}
+            <div class="matrixTxt">
+              <p>{matrixLeg}</p>
+            </div>
+          {/each}
+          {#each node.children as child, i}
+          
+            <!-- svelte-ignore a11y-click-events-have-key-events -->
+            <!-- svelte-ignore a11y-no-noninteractive-element-to-interactive-role -->
+            <div class="matrixTxt">
+              <SurveyNode
+                node={{ ...child.children[0], type: 'text' }}
+                {context}
+                {next} />
+            </div>
+            {#each matrix as nonused, j}
+              <div class:single={true && !className}>
+                <!-- svelte-ignore a11y-click-events-have-key-events -->
+                <!-- svelte-ignore a11y-no-noninteractive-element-to-interactive-role -->
+                <li
```

Comment:
What speaks against `input type=radio` (within the `li`s) also to make the a11y checker happier?

Reply:
Opted not to make this change at this moment. Should be part of a bigger change of all other types as well